sonolark/lib/kube/kube.go

```diff
@@ -731,3 +741,79 @@ func (m *kubePackage) kubeGet(ctx context.Context, r *apiResource, wait time.Dur
 
 	// not reachable
 }
+
+func (m *kubePackage) kubePortForwardFn(t *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if len(args) != 0 {
+		return nil, fmt.Errorf("<%v>: positional args not supported by `kube.portforward': %v", b.Name(), args)
+	}
+
+	if len(kwargs) < 3 {
+		return nil, fmt.Errorf("<%v>: expected at least <resource>=<name> <localport>=<port> <podport>=<port>", b.Name())
+	}
+
+	resource, name, err := getResourceAndName(kwargs[0])
+	if err != nil {
+		return nil, fmt.Errorf("<%v>: %s", b.Name(), err.Error())
+	}
+
+	// If resource is not namespace itself (special case), attempt to parse
+	// namespace out of the arg value.
+	var namespace string
+	if resource != namespaceResrc {
```

Comment:
Assuming this is just copy/paste? We'll just be doing port forwards for pods; it doesn't make sense to do it for a namespace or other resources does it?

Reply:
This is just copy/paste, port forwarding only applies to pods.